xivo_dao/resources/blocklist/persistor.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2025 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from sqlalchemy.orm import joinedload
+from xivo_dao.alchemy.blocklist import Blocklist, BlocklistNumber
+
+from xivo_dao.helpers.persistor import BasePersistor
+from xivo_dao.helpers.sequence_utils import split_by
+from xivo_dao.resources.utils.search import CriteriaBuilderMixin
+
+
+class Persistor(CriteriaBuilderMixin, BasePersistor):
+    _search_table = BlocklistNumber
+
+    def __init__(self, session, search_system, tenant_uuids=None):
+        self.session = session
+        self.search_system = search_system
+        self.tenant_uuids = tenant_uuids
+
+    def build_bulk_criteria(self, query, criteria):
+        for key, value in criteria.items():
+            assert key.endswith('_in')
```

Comment:
Do blocklists have a criteria ending with `_in`? I didn't find any

Reply:
that's a generic feature that was used in phone_numbers. Could extract it to a mixin instead of copying it in though.

Reply:
To be clear, how it works is that you can append `_in` to any model column (or "virtual" column) names in dao queries(`get`, `find_by`, `find_all_by`) in order to generate an `column IN (...)` clause on the query.